client/db.go

```diff
@@ -419,6 +419,19 @@ func (db *DB) Txn(retryable func(txn *Txn) error) error {
 	return newTxn(*db, 1 /* depth */).exec(retryable)
 }
 
+// BeginTxn starts a new transaction.
+func (db *DB) BeginTxn() *Txn {
+	return newTxn(*db, 1 /* depth */)
```

Comment:
client/db (which is really client/kv) doesn't need to know about `BeginTxn()`. I suggest you split a new method `NewTxn()` (which does the first bit (constructs the `Txn` and returns it)  from `newTxn()`, and use that in the sql code (with a small wrapper that sets its name to something useful like "sql:remoteaddr:...:starttime:....".

this is always called from the same place, so passing the depth into newTxn is not useful. I'd split `newTxn` into `newTxn` and `NewTxn`